### PR TITLE
feat: Description list plus extra operation area

### PR DESCRIPTION
**Pull request by zzz111**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### feat：描述列表加 extra 操作区域

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution
描述列表加 extra 操作区域

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Descriptions add `extra` prop. |
| 🇨🇳 Chinese | Descriptions 新增 extra 属性。 |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/descriptions/demo/basic.md](https://github.com/ant-design/ant-design/blob/wu-descriptions/components/descriptions/demo/basic.md)
[View rendered components/descriptions/index.en-US.md](https://github.com/ant-design/ant-design/blob/wu-descriptions/components/descriptions/index.en-US.md)
[View rendered components/descriptions/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/wu-descriptions/components/descriptions/index.zh-CN.md)

### CODE REVIEW

**Comment by zombieJ** [comment]
feature branch

File: components/descriptions/demo/basic.md [hunk 1]
```diff
@@ -14,10 +14,10 @@ title:
 Simplest Usage.
 
 ```jsx
-import { Descriptions } from 'antd';
+import { Descriptions, Button } from 'antd';
 
 ReactDOM.render(
-  <Descriptions title="User Info">
+  <Descriptions title="User Info" extra={<Button type="primary">Edit</Button>}>
```
Comment (zombieJ) [inline components/descriptions/demo/basic.md:20]: Move to `size.md` demo instead. And also update test case.

File: components/descriptions/index.en-US.md [hunk 2]
```diff
@@ -19,6 +19,7 @@ Commonly displayed on the details page.
 | Property | Description | Type | Default | Version |
 | --- | --- | --- | --- | --- |
 | title | The title of the description list, placed at the top | ReactNode | - |  |
+| extra | The action area of the description list, placed at the top-right | string \| ReactNode | - |  |
```
Comment (zombieJ) [inline components/descriptions/index.en-US.md:22]: Should add version info

File: components/descriptions/style/index.less [hunk 3]
```diff
@@ -5,11 +5,24 @@
 
 .@{descriptions-prefix-cls} {
   &-title {
+    display: inline-block;
```
Comment (zombieJ) [inline components/descriptions/style/index.less:8]: Do not modify origin title style. Add `-extra` is good enough.

File: components/descriptions/style/index.less [hunk 4]
```diff
@@ -5,11 +5,24 @@
...
+    max-width: 600px;
```
Comment (afc163) [inline components/descriptions/style/index.less:9]: image

空间够却显示不全。

File: components/descriptions/style/index.less [hunk 5]
```diff
@@ -5,11 +5,24 @@
...
     color: @heading-color;
     font-weight: bold;
     font-size: @font-size-lg;
     line-height: @line-height-base;
+    white-space: nowrap;
+    text-overflow: ellipsis;
+  }
+
+  &-extra {
+    float: right;
```
Comment (afc163) [inline components/descriptions/style/index.less:21]: image

应该缩略不挤压 extra 到第二行。

**Comment by afc163** [comment]
image

应该居中对齐。

File: components/descriptions/style/index.less [hunk 6]
```diff
@@ -4,12 +4,28 @@
...
     color: @heading-color;
     font-weight: bold;
     font-size: @font-size-lg;
     line-height: @line-height-base;
+    white-space: nowrap;
+    text-overflow: ellipsis;
+  }
+
+  &-extra {
+    margin-left: @descriptions-title-margin-bottom;
```
Comment (chenshuai2144) [inline components/descriptions/style/index.less:25]: 这个应该和 card 统一一下

File: components/descriptions/index.tsx [hunk 7]
```diff
@@ -148,7 +150,10 @@ function Descriptions({
       })}
       style={style}
     >
-      {title && <div className={`${prefixCls}-title`}>{title}</div>}
+      <div className={`${prefixCls}-header`}>
+        {(title || extra) && <div className={`${prefixCls}-title`}>{title}</div>}
+        {extra && <div className={`${prefixCls}-extra`}>{extra}</div>}
```
Comment (chenshuai2144) [inline components/descriptions/index.tsx:155]: 放个 space 包一下？这样可以自动给个 margin-right
@afc163 还是让用户自己选

File: components/descriptions/style/index.less [hunk 8]
```diff
@@ -4,12 +4,28 @@
...
 .@{descriptions-prefix-cls} {
-  &-title {
+  &-header {
+    display: flex;
+    align-items: center;
     margin-bottom: @descriptions-title-margin-bottom;
+  }
+
+  &-title {
+    flex: 1;
```
Comment (zombieJ) [inline components/descriptions/style/index.less:14]: `flex: auto`

File: components/descriptions/style/index.less (hunk 6 above)
Comment (zombieJ) [reply components/descriptions/style/index.less:25]: `@descriptions-title-margin-bottom` 不应该用来给 margin left 用

File: components/descriptions/style/index.less [hunk 9]
```diff
@@ -4,12 +4,28 @@
...
     font-size: @font-size-lg;
     line-height: @line-height-base;
+    white-space: nowrap;
+    text-overflow: ellipsis;
+  }
+
+  &-extra {
+    margin-left: @descriptions-title-margin-bottom;
+    color: @heading-color;
+    font-weight: normal;
```
Comment (zombieJ) [inline components/descriptions/style/index.less:27]: 这个应该用不到了

File: components/descriptions/index.tsx (hunk 7 above)
Comment (xrkffgg) [reply components/descriptions/index.tsx:155]: 我感觉用户自己选好一点，Space 的间隔自定义的话，还需要再加了。

File: components/descriptions/index.tsx [hunk 10]
```diff
@@ -148,7 +150,10 @@ function Descriptions({
       })}
       style={style}
     >
-      {title && <div className={`${prefixCls}-title`}>{title}</div>}
+      <div className={`${prefixCls}-header`}>
```
Comment (afc163) [inline components/descriptions/index.tsx:153]: title 和 extra 都没有的时候不显示 header

File: components/descriptions/index.tsx [hunk 11]
```diff
@@ -148,7 +150,10 @@ function Descriptions({
...
+        {(title || extra) && <div className={`${prefixCls}-title`}>{title}</div>}
```
Comment (afc163) [inline components/descriptions/index.tsx:154]: extra 和 title 应该关系不大。

Comment (chenshuai2144) [reply components/descriptions/index.tsx:154]: 这个用了 flex ，应该是为了让 extra 居左，不然就到了右边

Comment (zombieJ) [reply components/descriptions/index.tsx:154]: extra 设置 `margin-left: auto` 就行了，不需要占一个元素

**Comment by afc163** [comment]
ci broken

**Comment by zombieJ** [comment]
feature branch ood. Let me update feature first.

**Comment by zombieJ** [comment]
Merged. Rebase again~

**Review by afc163 — APPROVED** [review]

**Pull request merged**